app/src/test/java/tech/relaycorp/gateway/pdc/local/routes/ParcelCollectionRouteTest.kt

```diff
@@ -0,0 +1,114 @@
+package tech.relaycorp.gateway.pdc.local.routes
+
+import com.nhaarman.mockitokotlin2.any
+import com.nhaarman.mockitokotlin2.eq
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.verify
+import com.nhaarman.mockitokotlin2.whenever
+import io.ktor.http.cio.websocket.CloseReason
+import io.ktor.http.cio.websocket.Frame
+import io.ktor.http.cio.websocket.FrameType
+import io.ktor.http.cio.websocket.readReason
+import kotlinx.coroutines.flow.emptyFlow
+import kotlinx.coroutines.flow.flowOf
+import kotlinx.coroutines.test.runBlockingTest
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import tech.relaycorp.gateway.data.model.MessageAddress
+import tech.relaycorp.gateway.domain.endpoint.CollectParcels
+import tech.relaycorp.gateway.pdc.local.utils.ParcelCollectionHandshake
+import tech.relaycorp.gateway.test.WaitAssertions.suspendWaitFor
+import tech.relaycorp.relaynet.messages.control.ParcelDelivery
+import tech.relaycorp.relaynet.wrappers.x509.Certificate
+import javax.inject.Provider
+import kotlin.test.assertEquals
+import kotlin.test.assertTrue
+
+class ParcelCollectionRouteTest {
```

Comment:
I'd suggest adding tests for:

- Invalid Keep-Alive header value.
- Client closing the TCP connection abruptly. (Assuming we can do it from a unit test; otherwise we can do it in a functional/instrumentation test)
- Client closing the WebSocket connection with normal close code (`1000`)
- Client closing the WS connection with any close code but the normal one.
- Binary ACK messages.